### DISCUSSION TITLE
Executing a bash and a `expect` script as part of the same process in a container

### DISCUSSION THREAD

**Discussion by balvisio** [nextflow-io/nextflow#3041] [Q&A]
Hi all, I am trying to run both a bash script and a `expect` script sequentially in a container as part of a process. This what I am trying to achieve:
```
process my_process {
    container "ubuntu:20.04"

    script:
    template "my_script_1.sh"
    template "my_script_2.exp"
}
```

where `my_script_1.sh`:
```
#!/bin/bash
echo hello
```

and `my_script_2.exp`:
```
#!/usr/bin/expect -f
spawn ...
...
```

I have seen that given the code above, the container will only execute the second script only. Any help would be greatly appreciated. Any pointer to an example using this pattern would help a lot.

**Comment by jorgeaguileraseqera** [comment]
yes, you are right, only the last `template` is applied

at https://github.com/nextflow-io/nextflow/blob/7e8d2a5ab635b3577e4f6761df62a13863e5c40a/modules/nextflow/src/main/groovy/nextflow/processor/TaskRun.groovy#L661 nextflow parse the block and execute 2 times the 'template'  method who resolve the full path to the template but only the last Path is returning (because default groovy nature) and parsed

**Reply by balvisio** [reply]
Thank you for the answer! Is it possible to make the container execute both of them given that one is a bash script and the other an expect script?
